src/lib/adlib.js

```diff
@@ -170,22 +171,20 @@ Adlib.prototype.getURIFromPriref = async function(database, priref, type) {
         await sleep(timeout);
     }
     let querypath = `?output=json&database=${database}&search=priref=${priref}&limit=1`;
-    await (async () => {
-        // Get data from Redis cache.
-        let object = await redisClient.get(querypath);
-        if (object) {
-            object = JSON.parse(object);
-            console.log('Read from Redis.');
-        } else {
-            object = await this.fetchWithNTLM(querypath);
-            console.log('Read from Adlib.');
-            await redisClient.set(querypath, JSON.stringify(object));
-        }
+    //Get data from Redis cache.
+    let object = await redisClient.get(querypath);
+    if (object) {
+        object = JSON.parse(object);
+        console.log('Read from Redis.');
+    } else {
+        object = await this.fetchWithNTLM(querypath);
+        console.log('Read from Adlib.');
+        await redisClient.set(querypath, JSON.stringify(object));
```

Comment:
Set a timeout to redis object:
```suggestion
        await redisClient.setEx(querypath, 3600, JSON.stringify(object));
```